public_html/wp-content/plugins/pattern-directory/includes/pattern-validation.php

```diff
@@ -200,5 +204,141 @@ function validate_status( $prepared_post, $request ) {
+
+	// Extract URLs.
+	$links = array();
+	if ( preg_match_all( '![\b\W"\'](?P<link>((http|https|ftp|mailto):)?//[/.\w]+)[\b\W"\']!', $content, $m ) ) {
+		$links = array_unique( $m['link'] );
+	}
+
+	// Stringify.
+	if ( ! class_exists( '\WordPressdotorg\Pattern_Translations\Pattern' ) ) {
+		// This is just a fall-back for local environments where the Translator isn't active.
+		// not designed to be used in production.
+		$strings = array(
+			$title,
+			$description,
+			wp_strip_all_tags( $content ),
+			$keywords,
+		);
+	} else {
+		$pattern              = new Translations_Pattern();
+		$pattern->ID          = $post->ID;
+		$pattern->title       = $title;
+		$pattern->name        = $post->post_name;
+		$pattern->description = $description;
+		$pattern->keywords    = $keywords;
+		$pattern->html        = $content;
+		$pattern->locale      = get_locale();
+
+		$parser  = new Translations_PatternParser( $pattern );
+		$strings = $parser->to_strings();
+	}
+
+	// Combine strings for ease of use.
+	$combined_strings = implode( "\n", $strings );
+	$combined_links   = implode( "\n", $links );
+	$combined         = $combined_strings . "\n" . $combined_links;
+
+	// Not yet detected as spam.
+	$is_spam = false;
+
+	// Treat Paragraph-only submissions as likely spam.
+	if ( ! $is_spam ) {
+		// Only fetches the top-level of blocks, we're only
+		$block_names_in_use = array_filter(
+			array_unique(
+				wp_list_pluck(
+					parse_blocks( $content ),
+					'blockName'
+				)
+			)
+		);
+
+		if ( array( 'core/paragraph' ) === $block_names_in_use ) {
+			$is_spam = true;
+		}
+	}
+
+	// Run it past Akismet.
+	if ( ! $is_spam && is_callable( array( 'Akismet', 'rest_auto_check_comment' ) ) ) {
+		$current_user = wp_get_current_user();
+
+		$akismet_payload = array(
+			'comment_post_ID'      => 0,
+			'comment_type'         => 'pattern_submission',
+			// Disabled as logged in users get bonus points I think, which we don't want.
+			// 'user_ID'           => get_current_user_id(),
+			'comment_author'       => $current_user->display_name ?: $current_user->user_login,
+			'comment_author_email' => $current_user->user_email,
+			'comment_author_url'   => '',
+			'comment_content'      => $combined_strings,
+			'comment_content_raw'  => $content,
+			'permalink'            => get_permalink( $post ),
+		);
+
+		$akismet = \Akismet::rest_auto_check_comment( $akismet_payload );
+		if ( is_wp_error( $akismet ) ) {
+			$akismet = array( 'akismet_result' => 'discard' );
+		}
+
+		$is_spam = (
+			isset( $akismet['akismet_result'] ) &&
+			// true: spam, discard: 100% spam no-question.
+			( 'true' === $akismet['akismet_result'] || 'discard' === $akismet['akismet_result'] )
+		);
+	}
+
+	// Testing keyword. Case-sensitive.
+	if ( ! $is_spam && str_contains( $combined_strings, 'PatternDirectorySpamTest' ) ) {
+		$is_spam = true;
+	}
+
+	// If it's been detected as spam, flag it as pending-review.
+	if ( $is_spam ) {
+		$prepared_post->post_status = 'pending';
```

Comment:
Do you think we could also attach a note using the Internal Notes plugin? Otherwise it's not clear if this was pending on purpose (the default status is `pending`) vs spammed into pending.

Reply:
Ah, I had thought the plan was for auto-publish, not always going to pending.
Would it make more sense to add a different post_status for `spam` then?

Doesn't look like Internal Notes is in the local environment, so I didn't realise it was being used.

Reply:
>  I haven't been able to trigger Akismet spamming a pattern

I think about half of the seemingly-spam submissions to w.org/patterns triggered it (I copy pasted them over to local as a test), based on those posts I then added the `only-paragraphs` check.

I find it's harder to trigger Akismet with content than one would expect.. Looking at a public blogs spam comments should also pull up a few strings it'll trigger on.

Reply:
> Ah, I had thought the plan was for auto-publish, not always going to pending.

Clarification on that, I had assumed this due to everyone having `publish_pattern` capabilities from abb07a078e3041b14abef37ed1fa77f25f66618e.  Upon reviewing this, it seems like this method will prevent `publish`, so that's fine.

Reply:
> Would it make more sense to add a different post_status for spam then?

I've explored that in: https://github.com/dd32/pattern-directory/compare/add/pattern-validation...dd32:add/spam-status

Reply:
> Ah, I had thought the plan was for auto-publish, not always going to pending

It's an option, I don't know which we'll launch with. I think the plan is to eventually auto-publish, but be pending at launch to add a little bit of a manual quality check.

> Doesn't look like Internal Notes is in the local environment, so I didn't realise it was being used.

Oh oops, it should get added to the composer deps.

Reply:
> I find it's harder to trigger Akismet with content than one would expect.. Looking at a public blogs spam comments should also pull up a few strings it'll trigger on.

Finally got it to trigger, though I had to change the `comment_author` to a spammy name. The note added looks good 👍🏻

Reply:
> Would it make more sense to add a different post_status for spam then?

I like `add/spam-status`, I think that would make it much easier to separate out spam especially with everything pending at launch. Since you've already got it in a separate branch, would you rather make a separate PR for it once this lands, or combine it all?

Reply:
> I like add/spam-status, I think that would make it much easier to separate out spam especially with everything pending at launch.

Sounds good, I'll merge it into this PR.

> Since you've already got it in a separate branch

It was a branch forked from this branch, so I could quickly merge it into this branch if we went ahead with it.